src/partially_blindrsa.ts

```diff
@@ -0,0 +1,500 @@
+        const opts = { sLen: this.params.saltLength, hash };
+        const encoded_msg = await emsa_pss_encode(msg_prime, modulusLength - 1, opts);
+
+        // 4. m = bytes_to_int(encoded_msg)
+        const m = os2ip(encoded_msg);
+
+        // 5. c = is_coprime(m, n)
+        // 6. If c is false, raise an "invalid input" error
+        //    and stop
+        const c = is_coprime(m, n);
+        if (c === false) {
+            throw new Error('invalid input');
+        }
+
+        // 7. r = random_integer_uniform(1, n)
+        const r = random_integer_uniform(n, kLen);
+
+        // 8. inv = inverse_mod(r, n)
+        // 9. If inverse_mod fails, raise a "blinding error" error
+        //    and stop
+        let inv: Uint8Array;
+        try {
+            inv = i2osp(r.inverseMod(n), kLen);
+        } catch (e) {
+            throw new Error(`blinding error: ${(e as Error).toString()}`);
+        }
+
+        // 10. pk_derived = DerivePublicKey(pk, info)
+        const pk_derived = await this.derivePublicKey(pk, info);
+
+        // 11. x = RSAVP1(pk, r)
+        const x = rsavp1(pk_derived, r);
+
+        // 12. z = m * x mod n
+        const z = m.mulmod(x, n);
+
+        // 13. blind_msg = int_to_bytes(z, modulus_len)
+        const blindedMsg = i2osp(z, kLen);
+
+        // 14. output blind_msg, inv
+        return { blindedMsg, inv };
+    }
+
+    async blindSign(
+        privateKey: CryptoKey,
+        blindMsg: Uint8Array,
+        info: Uint8Array,
+    ): Promise<Uint8Array> {
+        const { jwkKey, modulusLengthBytes: kLen } = await this.extractKeyParams(
+            privateKey,
+            'private',
+        );
+        if (!jwkKey.n || !jwkKey.d) {
+            throw new Error('key has invalid parameters');
+        }
+        const n = sjcl.bn.fromBits(sjcl.codec.base64url.toBits(jwkKey.n));
+        const d = sjcl.bn.fromBits(sjcl.codec.base64url.toBits(jwkKey.d));
+        const p = sjcl.bn.fromBits(sjcl.codec.base64url.toBits(jwkKey.p));
+        const q = sjcl.bn.fromBits(sjcl.codec.base64url.toBits(jwkKey.q));
+        const sk: BigSecretKey = { n, d, p, q };
+
+        // 1. m = bytes_to_int(blind_msg)
+        const m = os2ip(blindMsg);
+
+        // 2. sk_derived, pk_derived = DeriveKeyPair(sk, info)
+        const { secretKey: sk_derived, publicKey: pk_derived } = await this.deriveKeyPair(sk, info);
+
+        if (this.params.supportsRSARAW) {
+            const { privateKey } = await PartiallyBlindRSA.bigKeyPairToCryptoKeyPair(
+                { secretKey: sk_derived, publicKey: pk_derived },
+                {
+                    modulusLength: kLen * 8,
+                    publicExponent: new Uint8Array([1, 0, 1]),
+                    hash: this.params.hash,
+                },
+                true,
+            );
+            return rsaRawBlingSign(privateKey, blindMsg);
+        }
+
+        // 3. s = RSASP1(sk_derived, m)
+        const s = rsasp1(sk_derived, m);
+
```

Comment:
I've done this both for Partially blind rsa and blind rsa (which presented the same behaviour). I've updated rsaRawBlingSign to return a sjcl.bn instead of a Uint8Array, for direct consumption by the algorithm.